Bryanna/visual admin dashboard

bry-gavino opened the pull request:
_Describe your feature here_

Made Admin Dash pages (I also accidentally made the Admin Mod page) along with functional links to help navigate. I'll guide through these links in the screenshots section.

I also made an AdminTable component. It's not functioning the way I want it to yet (still trying to figure out how to make it dynamic). The buttons I made for the table seem to break abstraction barriers to me, but I wanted to run it through you first.

## Screenshots

These are links which go to each page (separated by commas):
<img width="368" alt="Screen Shot 2020-10-21 at 5 08 16 PM" src="https://user-images.githubusercontent.com/60284767/96803029-0694dd80-13c0-11eb-8229-303fe4a93e08.png">

Accidentally made the Admin Mod :'DD (all good tho cuz it was fun):
<img width="1398" alt="Screen Shot 2020-10-21 at 5 09 03 PM" src="https://user-images.githubusercontent.com/60284767/96803080-23311580-13c0-11eb-8790-cdf892d5c2c1.png">

We can go back to the homepage by clicking on "NBJC" in the upper left corner:
<img width="177" alt="Screen Shot 2020-10-21 at 5 09 48 PM" src="https://user-images.githubusercontent.com/60284767/96803144-478cf200-13c0-11eb-9d1d-4c6c077a7289.png">

Admin Index page:
<img width="1397" alt="Screen Shot 2020-10-21 at 5 10 47 PM" src="https://user-images.githubusercontent.com/60284767/96803176-612e3980-13c0-11eb-80a2-9b7787939703.png">

We can click on these individual buttons to navigate to the other pages (except Questions--no page created for Questions yet):
<img width="231" alt="Screen Shot 2020-10-21 at 5 11 05 PM" src="https://user-images.githubusercontent.com/60284767/96803206-7acf8100-13c0-11eb-8aef-269987608170.png">

Organizations page:
<img width="1393" alt="Screen Shot 2020-10-21 at 5 11 51 PM" src="https://user-images.githubusercontent.com/60284767/96803220-8c188d80-13c0-11eb-972c-ca5f59a13883.png">

Users page (I didn't see a design for it on Figma so I left it black, but I made it fairly easy to implement any component--like AdminTable--into the page):
<img width="1398" alt="Screen Shot 2020-10-21 at 5 13 09 PM" src="https://user-images.githubusercontent.com/60284767/96803268-b66a4b00-13c0-11eb-8aa6-f6e4095da326.png">


CC: @kwfk

bry-gavino committed:
feat: added links to Admin Pages

links for guiding admin pages, not completely done or functional, pages have components in rightful
places

bry-gavino committed:
feat: added links to Admin Pages

links for guiding admin pages, not completely done or functional, pages have components in rightful
places

bry-gavino committed:
style: fixed minor link

bry-gavino committed:
docs: begin drawer

commit before making drastic changes

bry-gavino committed:
docs: begin drawer

commit before making drastic changes

bry-gavino committed:
feat: finished tables and drawer

made the table change due to the type of page and data being passed

bry-gavino committed:
feat: finished tables and drawer

made the table change due to the type of page and data being passed

bry-gavino commented:
<img width="1397" alt="Screen Shot 2020-10-28 at 5 49 04 AM" src="https://user-images.githubusercontent.com/60284767/97438237-c498f980-18e1-11eb-92ab-49106d352180.png">
<img width="1398" alt="Screen Shot 2020-10-28 at 5 48 46 AM" src="https://user-images.githubusercontent.com/60284767/97438239-c5319000-18e1-11eb-8d7a-8871f1c8cee8.png">
<img width="1390" alt="Screen Shot 2020-10-28 at 5 48 37 AM" src="https://user-images.githubusercontent.com/60284767/97438241-c5ca2680-18e1-11eb-9d78-ccfb9c841ab2.png">

kwfk commented on pages/adminDash/Organizations.tsx:
```diff
@@ -0,0 +1,12 @@
+import AdminIndex from 'components/adminDash/AdminIndex';
+import AdminTable from 'components/adminDash/AdminTable';
+// eslint-disable-next-line import/extensions
+import { sampleOrgData } from 'utils/sample-data.ts';
```
no `.ts` in import path
so you also won't need the eslint-disable

kwfk commented on pages/adminDash/Applications.tsx:
```diff
@@ -0,0 +1,12 @@
+import AdminIndex from 'components/adminDash/AdminIndex';
+import AdminTable from 'components/adminDash/AdminTable';
+// eslint-disable-next-line import/extensions
+import { sampleOrgAppData } from '../../utils/sample-data.ts';
```
no `.ts` in import path

kwfk commented on components/adminDash/AdminIndex.tsx:
```diff
@@ -0,0 +1,93 @@
+import Link from 'next/link';
+// eslint-disable-next-line import/no-extraneous-dependencies
```
don't need this eslint-disable anymore

kwfk commented on components/adminDash/AdminDrawer.tsx:
```diff
@@ -0,0 +1,97 @@
+import Drawer from '@material-ui/core/Drawer';
+import Hidden from '@material-ui/core/Hidden';
+import List from '@material-ui/core/List';
+// eslint-disable-next-line import/no-extraneous-dependencies
```
don't need eslint-disable

kwfk commented on components/adminDash/AdminDash.tsx:
```diff
@@ -0,0 +1,62 @@
+import Link from 'next/link';
+// eslint-disable-next-line import/no-extraneous-dependencies
```
don't need eslint-disable

kwfk commented on components/adminDash/AdminTable.tsx:
```diff
@@ -0,0 +1,137 @@
+/* eslint-disable import/no-extraneous-dependencies */
```
no need for eslint-disable

kwfk commented on pages/adminDash/Dashboard.tsx:
```diff
@@ -0,0 +1,41 @@
+import { Tab, Tabs, Button, IconButton } from '@material-ui/core';
+import AdminDash from 'components/adminDash/AdminDash';
+// eslint-disable-next-line import/no-extraneous-dependencies
```
no need for eslint-disable

kwfk commented on pages/adminDash/Users.tsx:
```diff
@@ -0,0 +1,12 @@
+import AdminIndex from 'components/adminDash/AdminIndex';
+import AdminTable from 'components/adminDash/AdminTable';
+// eslint-disable-next-line import/extensions
+import { sampleUserData } from 'utils/sample-data.ts';
```
no `.ts` in import path

kwfk commented on components/adminDash/AdminDrawer.tsx:
```diff
@@ -0,0 +1,97 @@
+                  <Button
+                    href="../adminDash/Applications"
+                    startIcon={<FiberManualRecordIcon />}
+                    variant={page === 'Application' ? 'contained' : ''}
+                    color={page === 'Application' ? 'primary' : ''}
```
use `undefined` instead of ''

kwfk replied:
and in all other similar places

kwfk reviewed (changes requested):
there are some common changes that should be easy fixes, but there are some other bigger changes that I think will be faster to talk to you
but it's looking great!

bry-gavino committed:
style: edited PR

removed dashboard layout, edited page names, wrapped buttons w links

bry-gavino committed:
style: wrapped buttons w links

bry-gavino committed:
style: git pulled this so i can push

kwfk commented on components/admin/AdminDrawer.tsx:
```diff
@@ -0,0 +1,103 @@
+import AdminTable from './AdminTable';
+import styles from '../../styles/admin/AdminDrawer.module.css';
+
+// try looking at Next.js documents to see if color can change depend on URL color
```
remove commetn

kwfk commented on components/admin/AdminIndex.tsx:
```diff
@@ -0,0 +1,103 @@
+          </Button>
+        </div>
+      </div>
+    </header>
```
don't actually need this because `Layout` is the navbar

kwfk commented on components/admin/AdminDrawer.tsx:
```diff
@@ -0,0 +1,103 @@
+    <main className={styles.content}>
+      <div className={styles.toolbar} />
+      {table}
+    </main>
```
instead of creating an empty space, we want to use flexbox in the parent to push the other content

kwfk commented on components/admin/AdminDrawer.tsx:
```diff
@@ -0,0 +1,103 @@
+// try looking at Next.js documents to see if color can change depend on URL color
+
+type Props = {
+  table: typeof AdminTable;
```
remove `table` because not being passed in

kwfk commented on components/admin/AdminDrawer.tsx:
```diff
@@ -0,0 +1,103 @@
+  page: string;
+};
+
+const AdminDrawer: React.FunctionComponent<Props> = ({ table, page }) => (
```
remove `table`

kwfk commented on components/admin/AdminIndex.tsx:
```diff
@@ -0,0 +1,103 @@
+              InputProps={{
+                startAdornment: (
+                  <InputAdornment position="start">
+                    <IconButton color="primary">
```
no button

kwfk commented on components/admin/AdminTable.tsx:
```diff
@@ -0,0 +1,138 @@
+    display: 'flex',
+    flexDirection: 'row',
+  },
+});
```
use css modules instead

kwfk commented on pages/admin/dashboard.tsx:
```diff
@@ -0,0 +1,60 @@
+import { useState } from 'react';
```
we can actually delete this file because we will use the same route as the moderator review

kwfk commented on styles/admin/AdminIndex.module.css:
```diff
@@ -0,0 +1,105 @@
+}
+
+.rightCol {
+  width: 80%;
```
this should be 100%

kwfk commented on components/admin/AdminTable.tsx:
```diff
@@ -0,0 +1,138 @@
+                  </TableCell>
+                </TableRow>
+              ))}
+          </TableBody>
```
use
```tsx
<caption>No {head} in database</caption>
```
where head is a new prop that can be 'organizations', 'users', or 'organization applications'

kwfk committed:
Merge branch 'main' into bryanna/visual-admin-dashboard

kwfk reviewed (changes requested):
it looks like the table doesn't differentiate between what page it's in? You can assume that the data you get is what you want to display on the table, but have it so that the column names are based on the keys of the data.
For example, if you given the data:
```javascript
{
    id: 1
    name: 'Fred Kim',
    work: 'coder',
    type: 'test',
    date: 'Oct. 31, 2020'
}
```
It will have the column names: id, name, work, type, date, but if the keys are different, the column names will change accordingly.
Let me know if you have questions about this, and I'll elaborate more.

make sure to remove the old components under AdminDash
also, remove any unused imports

bry-gavino committed:
feat: fixed table!

also added a little link in the layout just for the purpose of viewing the new features

kwfk committed:
refactor(admin dashboard): move components to match folder structure and change drawer style

Moved components so that it matches workspace convention. Change drawer to not use MUI Drawer and
make sure to use NextJS router

Clean stylesheet for admin drawer

kwfk commented on components/admin/AdminDrawer/index.tsx:
```diff
@@ -0,0 +1,73 @@
+import { Button, Hidden, List } from '@material-ui/core';
```
keep clean imports by using named exports like this for material ui

kwfk commented on components/admin/AdminIndex/AdminIndex.module.css:
```diff
@@ -0,0 +1,61 @@
+.layout {
```
remove unused styles

kwfk commented on components/admin/AdminTable/AdminTable.module.css:
```diff
@@ -0,0 +1,11 @@
+.table {
```
remove unused styles

kwfk commented on components/admin/AdminTable/index.tsx:
```diff
@@ -0,0 +1,129 @@
+
+const AdminTable: React.FunctionComponent<Props> = ({ data, pageType }) => {
+  const headList = Object.keys(data[0]);
+  const button1 =
+    // eslint-disable-next-line no-nested-ternary
+    pageType === 'applications'
+      ? declineButton
+      : pageType === 'organizations'
+      ? deleteButton
+      : suspendButton;
+  const button2 =
+    // eslint-disable-next-line no-nested-ternary
+    pageType === 'applications'
+      ? acceptButton
+      : pageType === 'organizations'
+      ? viewButton
+      : resetButton;
```
instead of disabling eslint use if statements instead

kwfk commented on styles/admin/AdminDash.module.css:
```diff
@@ -0,0 +1,54 @@
+.layout {
```
delete if unused

kwfk commented on styles/admin/AppIndex.module.css:
```diff
@@ -0,0 +1,47 @@
+.page {
```
delete if unused

kwfk commented on styles/admin/AppReview.module.css:
```diff
@@ -0,0 +1,51 @@
+.content {
```
delete if unused

kwfk reviewed (changes requested):
just some cleaning up left!
I committed some changes to the drawer and refactored the components so they match the folder structure (page-specific components should go inside a folder with its stylesheet)

I'll add additional comments about the type errors on AdminTable

kwfk commented on components/admin/AdminTable/index.tsx:
```diff
@@ -0,0 +1,129 @@
+);
+
+type Props = {
+  data: Array<unknown>;
```
use `OrgApp | Organization | User` imported from `interfaces`

bry-gavino committed:
style: cleaning

bry-gavino committed:
style: cleaning

kwfk commented on components/admin/AdminTable/index.tsx:
```diff
@@ -0,0 +1,139 @@
+  pageType: string;
+};
+
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+function giveButton1(pT: string) {
+  if (pT === 'applications') {
+    return declineButton;
+  }
+  if (pT === 'organizations') {
+    return deleteButton;
+  }
+  return suspendButton;
+}
+
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+function giveButton2(pT: string) {
+  if (pT === 'applications') {
+    return acceptButton;
+  }
+  if (pT === 'organizations') {
+    return viewButton;
+  }
+  return resetButton;
+}
```
move this inside the component body and use the return type `React.ReactElement`
you can also use arrow functions like
```typescript
const button = () => {...}
```
inside the function, you also don't have pass in an param bc you have access to pageType

kwfk reviewed (changes requested):
last change :D

kwfk commented on components/admin/AdminTable/index.tsx:
```diff
@@ -0,0 +1,139 @@
+);
+
+type Props = {
+  data: Array<OrgApp | Organization | User>;
```
remember to import from interfaces

kwfk committed:
fix(admin index): keep consistent prop propagation with page prop

rename head prop to page prop for all admin components

bry-gavino committed:
feat: fixed component buttons functions

kwfk committed:
Merge branch 'bryanna/visual-admin-dashboard' of github.com:calblueprint/nbjc into bryanna/visual-admin-dashboard

kwfk committed:
refactor(admin): small refactoring to clean up code

kwfk committed:
refactor(admin dash css): remove old adminDash css

kwfk reviewed (approved):
looks good! nice work with all the revisions XD
